src/core/unit.h

```diff
@@ -621,6 +621,9 @@ typedef struct UnitVTable {
 
         /* True if queued jobs of this type should be GC'ed if no other job needs them anymore */
         bool gc_jobs:1;
+
+        /* True if systemd-oomd can monitor and kill this unit's recursive children's cgroup(s)  */
+        bool can_do_managed_oom_kill:1;
```

Comment:
I'll update to add this PR to add services and scopes now that the slices stuff is sufficiently tested. Logically slices seemed the best suited as a first step since they're designed for cgroup tree groupings.

There could be cases where you have a protected workload (e.g. hostcritical.slice) and you don't want systoomd to act on that slice, but you want systoomd to act on user.slice so opting in/out the entire tree is too restrictive. I'm also working on an interface update where the thresholds are more configurable since each subtree has a different tolerance for pressure; while you could enable it for the whole tree in that case it would be more useful to set lower thresholds for user.slice where actions can be taken more eagerly.